set/name_trie.go

```diff
@@ -0,0 +1,96 @@
+package set
+
+import (
+    "sort"
+)
+
+type NameTrie struct {
```

Comment:
Very usecase-specific name for something that is otherwise very general to any strings.